apps/core/views.py

```diff
@@ -412,3 +413,36 @@ def post(self, request, *args, **kwargs):
         user.save()
 
         return Response('Your password has been changed')
+
+
+class CheckEmailExistsView(APIView):
+    """
+    This view checks if a user with a certain email exists in the database.
+    """
+    queryset = User.objects.all()
+    permission_classes = (IsAuthenticated,)
```

Comment:
this endpoint must be used before registration, so we can't check if the user is authorised

Reply:
fixed in a new commit